### PR TITLE
Optimize method calls w props receiver

### PR DESCRIPTION
**Pull request by josephsavona**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #31771

### CODE REVIEW

**Comment by josephsavona** [comment]
Closes #31763

File: compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/jsx-outlining-child-stored-in-id.expect.md [hunk 1]
```diff
@@ -50,26 +50,35 @@ export const FIXTURE_ENTRYPOINT = {
 ```javascript
 import { c as _c } from "react/compiler-runtime"; // @enableJsxOutlining
 function Component(arr) {
```
Comment (mofeiZ) [inline compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/jsx-outlining-child-stored-in-id.expect.md:52]: nit: can we rewrite this fixture to destructure props i.e. `function Component({ arr })`?

File: compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/props-method-dependency.expect.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  } else {
+    t1 = $[3];
+  }
+  return t1;
+}
+
+```
+      
+### Eval output
+(kind: exception) Fixture not implemented
```
Comment (mofeiZ) [inline compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/props-method-dependency.expect.md:41]: nit: could we add a `FIXTURE_ENTRYPOINT` export here?

**Review by mofeiZ — APPROVED** [review]
nice!!

**Pull request closed without merging**